specs/data_structures.md

```diff
@@ -105,23 +105,29 @@ Blocks are the top-level data structure of the LazyLedger blockchain.
+| `lastCommitRoot`                  | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                                                                                                                           |
+| `consensusRoot`                   | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block.                                                                                                       |
+| `stateCommitment`                 | [HashDigest](#hashdigest) | The [state root](#state) after this block's transactions are applied.                                                                                                              |
+| `availableDataOriginalSharesUsed` | `uint64`                  | The number of shares used in the [original data square](#arranging-available-data-into-shares) that are not [tail padding](./consensus.md#reserved-namespace-ids).                 |
+| `availableDataOriginalSquareSize` | `uint64`                  | The number of rows/columns of the original data [shares](data_structures.md#share) in [square layout](#arranging-available-data-into-shares) for this block. Must be a power of 2. |
+| `availableDataRoot`               | [HashDigest](#hashdigest) | Root of [commitments to erasure-coded data](#availabledataheader).                                                                                                                 |
+| `proposerAddress`                 | [Address](#address)       | Address of this block's proposer.                                                                                                                                                  |
 
 ### AvailableDataHeader
 
-| name                       | type                          | description                            |
-| -------------------------- | ----------------------------- | -------------------------------------- |
-| `availableDataCommitments` | [HashDigest](#hashdigest)`[]` | Commitments to all erasure-coded data. |
+| name       | type                          | description                            |
+| ---------- | ----------------------------- | -------------------------------------- |
+| `rowRoots` | [HashDigest](#hashdigest)`[]` | Commitments to all erasure-coded data. |
+| `colRoots` | [HashDigest](#hashdigest)`[]` | Commitments to all erasure-coded data. |
+
+The `availableDataRoot` of the [header](#header) is computed using the row and column roots as described [here](#2d-reed-solomon-encoding-scheme).
+
+The number of column roots is twice the [header](#header)'s `availableDataOriginalSquareSize` field, and the number of row roots is `ceil(availableDataOriginalSharesUsed / availableDataOriginalSquareSize) + availableDataOriginalSquareSize`.
```

Comment:
So #column roots = 2 * availableDataOriginalSquareSize because it contains all the parity shares, too. And for the row roots, we only compute the roots that do not only contain tail padding. Maybe add the latter as a sentence before or after this?

Reply:
Fixed in 64a08e7.